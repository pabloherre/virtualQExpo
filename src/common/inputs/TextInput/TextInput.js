import React, {Component} from 'react';
import {View, Text, TextInput as NativeTextInput} from 'react-native';

class TextInput extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.label}>{this.props.label}</Text>
        <NativeTextInput style={styles.input} {...this.props} />
      </View>
    );
  }
}

const styles = {
  container: {
    paddingTop: 10,
    paddingBottom: 10,
  },
  input: {
    height: 40,
    fontSize: 20,
    borderBottomColor: '#7DD7F1',
    borderBottomWidth: 1,
  },
  label: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#7DD7F1',
  },
};

export default TextInput;
