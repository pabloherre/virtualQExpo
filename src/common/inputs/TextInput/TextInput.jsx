import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, Text, TextInput as NativeTextInput } from 'react-native';
import styles from './TextInput.styles';

class TextInput extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { label, ...rest } = this.props;
    return (
      <View style={styles.container}>
        <Text style={styles.label} {...setTestInfo('testTextInputLabel')}>
          {label}
        </Text>
        <NativeTextInput style={styles.input} {...rest} {...setTestInfo('testInputInput')} />
      </View>
    );
  }
}

TextInput.propTypes = {
  label: PropTypes.string.isRequired
};

export default TextInput;
