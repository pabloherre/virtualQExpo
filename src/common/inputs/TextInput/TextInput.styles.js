import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    paddingTop: 10,
    paddingBottom: 10
  },
  input: {
    height: 40,
    fontSize: 20,
    borderBottomColor: '#7DD7F1',
    borderBottomWidth: 1
  },
  label: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#7DD7F1'
  }
});
