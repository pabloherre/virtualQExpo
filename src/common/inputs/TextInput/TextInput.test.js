import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TextInput from './TextInput';

describe('<TextInput />', () => {
  it('should render correctly', async () => {
    const textInput = renderer.create(<TextInput label="Test" />);

    expect(textInput).toMatchSnapshot();
  });

  it('should render the passed label', async () => {
    const wrapper = shallow(<TextInput label="Test" />);
    const label = wrapper.childAt(0);
    expect(label.contains('Test')).toBe(true);
  });

  it.todo('should validate email format if type="email"');
});
