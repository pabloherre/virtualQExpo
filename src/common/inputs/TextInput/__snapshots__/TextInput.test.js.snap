// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render correctly 1`] = `
<View
  style={
    Object {
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#7DD7F1",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Test
  </Text>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomColor": "#7DD7F1",
        "borderBottomWidth": 1,
        "fontSize": 20,
        "height": 40,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
